python/equistore-operations/equistore/operations/_dispatch.py

```diff
@@ -98,9 +98,12 @@ def bincount(input, weights=None, minlength=0):
 
 
 def array_like_data(array, data):
-    """Return `data` with the same data-type and array-type of array"""
-    if isinstance(array, TorchTensor):
-        _check_all_torch_tensor([data])
+    """Create an object with the same type as `array` containing the given `data`."""
+    if type(array) is type(data):
+        return data
```

Comment:
This is still missing `dtype` conversion. I'm not sure why this change is needed, could you explain?

Reply:
This is weird code indeed. Let me change this back to something useful. The problem is that it does not work if data is a `list`.